jax/interpreters/masking.py

```diff
@@ -311,13 +311,17 @@ def finalize_spec(polymorphic_shape, padded_shape):
                for e, d in zip(polymorphic_shape, padded_shape))
 
 def parse_spec(spec=''):
-  if not spec:
-    return ShapeSpec(())
-  if spec[0] == '(':
-    if spec[-1] != ')': raise ShapeSyntaxError(spec)
-    spec = spec[1:-1]
-  dims = map(_parse_dim, spec.replace(' ', '').strip(',').split(','))
-  return ShapeSpec(dims)
+  try:
```

Comment:
There are two changes here, the string "()" was not processed properly, and I wanted an error message that includes the original in_shapes not just the part that failed. 

I guess it makes sense to add some masking_test.py, but I don't know what is the status of that file.

Reply:
I have reverted this change